src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogGraphListener.java

```diff
@@ -103,13 +106,43 @@ public void onNewHead(FlowNode flowNode) {
             TagsUtil.addTagToTags(tags, "stage_depth", String.valueOf(stageDepth));
             // Add custom result tag
             TagsUtil.addTagToTags(tags, "result", result);
+            long pauseDuration = getPauseDuration(startNode);
+
             client.gauge("jenkins.job.stage_duration", getTime(startNode, endNode), hostname, tags);
+            client.gauge("jenkins.job.stage_pause_duration", pauseDuration, hostname, tags);
             client.incrementCounter("jenkins.job.stage_completed", hostname, tags);
         } catch (IOException | InterruptedException e) {
             DatadogUtilities.severe(logger, e, "Unable to submit the stage duration metric for " + getStageName(startNode));
         }
     }
 
+    private long getPauseDuration(@Nonnull FlowNode startNode) {
```

Comment:
for consistency with the FlowNode API
```suggestion
    private long getPauseDurationMillis(@Nonnull FlowNode startNode) {
```

Reply:
And for better readability 👍